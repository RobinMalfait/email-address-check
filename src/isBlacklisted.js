const dnsbl = require("dnsbl");
import getIPAddress from "./getIPAddress";

async function isBlacklisted(email) {
  const [username, hostname] = email.split("@");

  try {
    const ip = await getIPAddress(hostname);
    return await dnsbl.lookup(ip, "zen.spamhaus.org");
  } catch (err) {
    return false;
  }
}

export default isBlacklisted;
