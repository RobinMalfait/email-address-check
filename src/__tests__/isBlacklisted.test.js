import isBlacklisted from "../isBlacklisted";

describe("isBlacklisted", () => {
  it("should mark the email as blacklisted", () => {
    return expect(
      isBlacklisted("example@[98.150.108.228]")
    ).resolves.toMatchSnapshot();
  });
});
