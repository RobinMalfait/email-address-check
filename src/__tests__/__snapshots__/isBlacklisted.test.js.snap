// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isBlacklisted should mark the email as blacklisted 1`] = `true`;
